extensions/agroal/deployment/src/test/java/io/quarkus/agroal/test/AgroalVaultITCase.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkus.agroal.test;
+
+import static io.quarkus.vault.VaultTestExtension.testDataSource;
+
+import java.sql.SQLException;
+
+import javax.inject.Inject;
+
+import org.jboss.logging.Logger;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.agroal.api.AgroalDataSource;
+import io.quarkus.agroal.DataSource;
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.vault.VaultTestExtension;
+
+@DisabledOnOs(OS.WINDOWS)
+@ExtendWith(VaultTestExtension.class)
+public class AgroalVaultITCase {
+
+    private static final Logger log = Logger.getLogger(AgroalVaultITCase.class.getName());
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest().setArchiveProducer(
+            () -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource("application-vault-datasource.properties", "application.properties"));
+
+    //tag::injection[]
+    @Inject
+    @DataSource("staticDS")
+    AgroalDataSource staticDS;
+
+    @Inject
+    @DataSource("dynamicDS")
+    AgroalDataSource dynamicDS;
+    //end::injection[]
```

Comment:
I suppose you will use source includes in the doc at some point in this PR: bad luck, it doesn't work on the website so you will need to get rid of them.